connection_go18.go

```diff
@@ -0,0 +1,24 @@
+// +build go1.8
+
+package mysql
+
+import (
+	"context"
+)
+
+func (mc *mysqlConn) Ping(ctx context.Context) error {
+	err := mc.writeCommandPacket(comPing)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case <-ctx.Done():
+		mc.Close()
+		return ctx.Err()
+	default:
+	}
+
+	_, err = mc.readResultOK()
```

Comment:
FYI, this is how to interrupt `conn.Read()`.
https://github.com/cybozu-go/cmd/blob/21e6f5953122316b450289d4e20fb664d84ffe41/http.go#L226
(thanks to @ymmt2005)